src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases import OrganizationEventsV2EndpointBase
+from sentry.models import Project, ProjectStatus
+from sentry.api.bases.organization import OrganizationPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.discover.models import KeyTransaction, MAX_KEY_TRANSACTIONS
+from sentry.snuba.discover import query
+
+
+class KeyTransactionEndpoint(OrganizationEventsV2EndpointBase):
+    permission_classes = (OrganizationPermission,)
+
+    def has_feature(self, request, organization):
+        return features.has("organizations:discover", organization, actor=request.user)
+
+    def get_project(self, organization, project_id):
+        """ Get a project for an org, or None if one doesn't exist """
+        return Project.objects.filter(
+            organization=organization, status=ProjectStatus.VISIBLE, id=project_id
+        ).first()
```

Comment:
This isn't checking any access/team assignments. You should use `OrganizationEndpoint.get_projects()` instead and provide it a list of the only project you want. That will ensure that we continue to apply project permissions consistently with other endpoints.